draft-ietf-tls-esni.md

```diff
@@ -430,6 +430,35 @@ the ClientHelloOuter. The server then computes the digest of the reconstructed
 ClientHelloInner. If the digest does not equal OuterExtensions.inner_digest,
 then the server MUST abort the connection with a "decrypt_error" alert.
 
+# The "handshake_padding" Extension {#handshake-padding}
+
+Encrypting the ClientHelloInner may not be sufficient to protect sensitive
+parameters. In particular, the length of the ClientHelloInner, or the length of
+the backend server's response, may leak information about these parameters. The
+client and backend server can use the "handshake_padding" extension in order to
+mitigate this leakage.
+
+A new extension type is defined, which the client MAY include in its
+ClientHello.
+
+~~~
+    enum {
+       handshake_padding(0xff05), (65535)
+    } ExtensionType;
+~~~
+
+When solicited by the client, the server MAY reply by including the extension in
```

Comment:
When *and only when* solicited by the client? If so, why?

Reply:
Yes, due to the rule in TLS !.3 that the server not send unsolicited extensions (with a few, well-specified exceptions). There'd be no need to allow this behavior anyway: if the client failing to pad leads to a confidentiality breach, it's unlikely that the server can mitigate this much by padding unilaterally.